simple/reprovide_test.go

```diff
@@ -9,11 +9,12 @@ import (
 	"github.com/ipfs/go-cid"
 	ds "github.com/ipfs/go-datastore"
 	dssync "github.com/ipfs/go-datastore/sync"
-	"github.com/ipfs/go-ipfs-blockstore"
+	bsfetcher "github.com/ipfs/go-fetcher/impl/blockservice"
+	blockstore "github.com/ipfs/go-ipfs-blockstore"
 	offline "github.com/ipfs/go-ipfs-exchange-offline"
 	mock "github.com/ipfs/go-ipfs-routing/mock"
 	cbor "github.com/ipfs/go-ipld-cbor"
-	merkledag "github.com/ipfs/go-merkledag"
+	_ "github.com/ipld/go-ipld-prime/codec/dagcbor"
```

Comment:
Is this because our test objects are constructed using go-ipld-cbor instead of go-ipld-prime's dagcbor codec? Can we just drop go-ipld-cbor as a dependency here?

Reply:
This is probably fine and correct.  The current way we handle registration of codec implementations is by side-effecting imports (the init method in the package registers a multicodec in a global table).

I'm not "thrilled" with this per se, because it's very side-effect heavy (and I'm definitely a functional programmer at heart), but it's a race-condition-free way to solve a problem, and seems to be one of the least-bad options available to us.

Reply:
good news, we can drop the blank import now since we already import `dagcbor` and we're importing twice now.